components/TaskCard/TaskCard.js

```diff
@@ -1,26 +1,151 @@
 import styled from "styled-components";
+import { useState } from "react";
+import { css } from "styled-components";
+
+import AddTaskForm from "@/pages/addTask";
+
+export default function TaskCard({ task, updateTask }) {
+  const [editMode, setEditMode] = useState(false);
+  const [taskout, setTaskOut] = useState({ ...task });
```

Comment:
I don't think that you need to spread the `task` when initializing the state:
```suggestion
  const [taskout, setTaskOut] = useState(task);
```